analysis/study_definition.py

```diff
@@ -1,13 +1,884 @@
+          "incidence": 0.01,
+      },
+  ) 
+def covid_therapeutics(dx_codelist):
+  return patients.with_covid_therapeutics(
+      dx_codelist,
+      with_these_indications = "non_hospitalised",
+      between = ["covid_test_positive_date", "covid_test_positive_date + 5 days"],
+      find_first_match_in_period=True,
+      returning="date",
+      date_format="YYYY-MM-DD",
+      return_expectations={
+          "incidence": 0.9,
+          "date": {"earliest": "2021-12-16"},
+      },
+  )
+def adverse_outcome_icd(dx_codelist):
+  return patients.admitted_to_hospital(
+      with_these_diagnoses = dx_codelist,
+      returning="date_admitted",
+      between = ["covid_test_positive_date", "covid_test_positive_date + 28 days"],
+      find_first_match_in_period=True,
+      date_format="YYYY-MM-DD",
+      return_expectations={
+          "incidence": 0.2,
+          "date": {"earliest": "2021-12-16"},
+      },
+  )
+def adverse_outcome_snomed(dx_codelist):
+  return patients.with_these_clinical_events(
+      dx_codelist,
+      returning="date",
+      between = ["covid_test_positive_date", "covid_test_positive_date + 28 days"],
+      find_first_match_in_period=True,
+      date_format="YYYY-MM-DD",
+      return_expectations={
+          "incidence": 0.2,
+          "date": {"earliest": "2021-12-16"},
+      },
+  )
+def imae_snomed(dx_codelist):  ## to idenitfy NEW conditions, will include all first diagnosis but then on stata remove those diagnosed before covid test positive date
+  return patients.with_these_clinical_events(
+      dx_codelist,
+      returning="date",
+      on_or_before ="covid_test_positive_date + 28 days",
+      find_first_match_in_period=True,
+      date_format="YYYY-MM-DD",
+      return_expectations={
+          "incidence": 0.2,
+          "date": {"earliest": "2021-12-16"},
+      },
+  )
+def imae_icd(dx_codelist):  ## to idenitfy NEW conditions, will include all first diagnosis but then on stata remove those diagnosed before covid test positive date
+  return patients.admitted_to_hospital(
+      with_these_diagnoses = dx_codelist,
+      returning="date_admitted",
+      on_or_before ="covid_test_positive_date + 28 days",
+      find_first_match_in_period=True,
+      date_format="YYYY-MM-DD",
+      return_expectations={
+          "incidence": 0.2,
+          "date": {"earliest": "2021-12-16"},
+      },
+  )
 study = StudyDefinition(
-    default_expectations={
-        "date": {"earliest": "1900-01-01", "latest": "today"},
-        "rate": "uniform",
-        "incidence": 0.5,
+  # Configure the expectations framework
+  default_expectations = {
+    "date": {"earliest": "2021-11-01", "latest": "today"},
+    "rate": "uniform",
+    "incidence": 0.05,
+  },
+      
+  ## Define index date
+  index_date = campaign_start,
+
+  population = patients.satisfying(
+    """
+    age >= 18 AND age < 110
+    AND NOT has_died
+    AND covid_test_positive
+    AND eligible_binary
+    """,
+  ),
```

Comment:
So as to avoid excluding a small number of individuals who will have been given a therapeutic but might be missing some data (i.e. a recorded positive covid test) I think you need to update to:

```
  population = patients.satisfying(
    """
    age >= 12 AND age < 110
    AND NOT has_died
    AND (
        registered_eligible
        AND
        high_risk_cohort_covid_nhsd
        AND
        covid_test_positive
        ) 
      OR 
        (  
        registered_treated 
        AND
        high_risk_cohort_covid_therapeutics
        )
    """,
    
    registered_eligible = patients.registered_as_of("covid_test_positive_date"),
    registered_treated = patients.registered_as_of("date_treated"),
    
  ),```

where `high_risk_cohort_covid_nhsd` is the same as `eligible` on line 444 (so rename `eligible` to `high_risk_cohort_covid_nhsd`?) 

Might be worth discussing this in person to ensure this makes sense for what you want. It might also be worth considering adding `AND NOT covid_positive_previous_30_days`

Reply:
Yes this is a good idea, as i do want to catch all patients who have had a therapeutic but might be missing a recorded positive covid test). 

However `high_risk_cohort_covid_nhsd` is not the same as `eligible`. I have created my own eligible group based. I then combined them with 'high_risk_cohort_covid_nhsd' to try to pick up everyone that would be defined as eligible, including those in the control group who did not get a covid therapeutic but were eligible, Is this not the correct thing to do? 

Maybe we could discuss the high_risk_cohort_covid_nhsd??

Reply:
The open codelists that i have used to generate my eligible (high_risk_cohort_covid_nhsd) cohort